tests/lib/cmsis_dsp/transform/src/cf64.c

```diff
@@ -167,42 +167,42 @@ DEFINE_TEST_VARIANT4(arm_cifft_f64_cmplx, step_4096, &arm_cfft_sR_f64_len4096,
-		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_2048),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_2048),
 		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_4096),
 		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_4096),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_16),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_16),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_32),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_32),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_64),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_64),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_128),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_128),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_256),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_256),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_512),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_512),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_1024),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_1024),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_2048),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_2048),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_1024),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_1024),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_512),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_512),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_256),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_256),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_128),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_128),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_64),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_64),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_32),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_32),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_noisy_16),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_noisy_16),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_4096),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_4096),
 		ztest_unit_test(test_arm_cfft_f64_cmplx_step_2048),
 		ztest_unit_test(test_arm_cifft_f64_cmplx_step_2048),
-		ztest_unit_test(test_arm_cfft_f64_cmplx_step_4096),
-		ztest_unit_test(test_arm_cifft_f64_cmplx_step_4096)
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_1024),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_1024),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_512),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_512),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_256),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_256),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_128),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_128),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_64),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_64),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_32),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_32),
+		ztest_unit_test(test_arm_cfft_f64_cmplx_step_16),
+		ztest_unit_test(test_arm_cifft_f64_cmplx_step_16)
```

Comment:
From #27849, re:
> I don't know exactly how malloc() algorithm from newlibc is working,
> but it appears during my debug session that even if memory is freed between each step,
> each time the size of the malloc is increased, the previous memory size is 'lost' (probably due to heap fragmentation).

IMO, this needs more investigation. I am not sure if fragmentation is the cause as far as this test goes (there aren't all that many mixed-size mallocs here to begin with -- all the allocated blocks are freed before subsequent different-size mallocs); rather, it seems there is something strange going on.

Reply:
IMO fragmentation is the issue:
When test was in increasing order, the return of malloc() was respectively:
0x200017A8
0x200018B0
0x20001AB8
0x20001EC0
0x200026C8
0x200036D0 (step 512, malloc 0x2000)
0x200056D8 (step 1024, malloc 0x4000)
0x200096E0  (step 2048, malloc 0x8000)
Each time the malloc pointer is a bit more than the previous malloc.
I can just assume maaloc algo can reuse freed chunck for smaller malloc but cannot reuse for lager one.

Reply:
@stephanosio ^^

Reply:
Fair enough.